packages/edit-site/src/components/revisions/index.js

```diff
@@ -25,13 +25,29 @@ const {
 	ExperimentalBlockEditorProvider,
 	GlobalStylesContext,
 	useGlobalStylesOutputWithConfig,
+	useGetBlockVariationStylesWithConfig,
 } = unlock( blockEditorPrivateApis );
 const { mergeBaseAndUserConfigs } = unlock( editorPrivateApis );
 
 function isObjectEmpty( object ) {
 	return ! object || Object.keys( object ).length === 0;
 }
 
+const cloneObject = ( obj ) => JSON.parse( JSON.stringify( obj ) );
+function RevisionStyles( { styles, config } ) {
+	const variationOverrides = useSelect(
+		( select ) => unlock( select( blockEditorStore ) ).getStyleOverrides(),
+		[ config ]
+	);
+
+	const mergedOverrides = useGetBlockVariationStylesWithConfig(
+		config,
+		cloneObject( variationOverrides )
```

Comment:
Otherwise we're mutating the overrides object it seems. If we don't clone, we can just run the hook and not pass the result to Editor Styles.

This seems like a bug in the store, because mutating shouldn't be possible? That is, the result from useSelect should always be a unique reference. 

I don't know yet. Too hungry. I'll check.